src/Persistence/Proxy/CreatedObjectsTracker.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Zenstruck\Foundry\Persistence\Proxy;
+
+use WeakReference;
+
+use function Zenstruck\Foundry\Persistence\refresh;
+
+final class CreatedObjectsTracker
+{
+    /** @var list<\WeakReference<object>> */
+    private static $buffer = [];
+
+    public static function add(object $object): void
+    {
+        self::$buffer[] = \WeakReference::create($object);
```

Comment:
using `WeakReference` allows to not track too much objects

Reply:
Not WeakMap?

Reply:
How I understand it, `WeakMap` is used when you need the object to be the _key_ in the dictionnary

what I'd need is a `WeakList`, but this does not exist, so I'm dealing with a list of `WeakReference`

Reply:
Ah, got it!